lang/stack/Portfile

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        commercialhaskell stack 2.1.1
+
+name                stack
+categories          lang haskell
+maintainers         {ieee.org:s.t.smith @essandess} openmaintainer
+license             BSD
+platforms           darwin
+
+description The Haskell Tool Stack
+long_description    ${description}. \
+        Stack is a cross-platform command line interface for developing Haskell projects. \
+        It features: installing GHC automatically, installing packages needed for your project, \
+        building your project, testing your project, benchmarking your project. \
+        It is intended for Haskellers both new and experienced.
+
+homepage            https://haskellstack.org
+
+variant bootstrap \
+	    description {Bootstrap a build of stack.} {}
+
+# make bootstrapping the default when/if it works
+default_variants    -bootstrap
+
+master_sites        ${github.homepage}/releases/download/v${github.version}
+distfiles           ${name}-${github.version}-osx-x86_64${extract.suffix} \
+                    ${name}-${github.version}-osx-x86_64${extract.suffix}.sha256
+distname            ${name}-${github.version}-osx-x86_64
+
+set sha256_checksum f4af329419fb6ee9655b22db04d72a35a5a225e78bdcc605d78334a72c8c2332
+checksums           ${name}-${github.version}-osx-x86_64${extract.suffix} \
+                    rmd160  45c8fc4537bf3fdd46854dcd56d51e4c041ac118 \
+                    sha256  ${sha256_checksum} \
+                    size    8316888 \
+                    ${name}-${github.version}-osx-x86_64${extract.suffix}.sha256 \
+                    rmd160  797373009deba25b55705ec275a5242e5507ed22 \
+                    sha256  028dc67190c4a4154b2ad7cd48a869235f618d56f36346dcf1b4d367b4fbf898 \
+                    size    65
+
+if {[variant_isset "bootstrap"]} {
+    master_sites-append \
+                    ${github.homepage}/archive
+    distfiles-append \
+                    v${github.version}${extract.suffix}
+    checksums-append \
+                    v${github.version}${extract.suffix} \
+                    rmd160  60f7fcc5548aabec357be217d11d0d1d616c1806 \
+                    sha256  10c0119456748b189724ee42dea093c324d101487c3d75d57eb625675bb57424 \
+                    size    816217
+}
+
+supported_archs     x86_64
+
+pre-extract {
+    # second SHA256 comparison against stack-provided hash
+    set sha256_verification [exec /bin/sh -c "\[\[ \$(< ${distpath}/${name}-${github.version}-osx-x86_64${extract.suffix}.sha256) == \"${sha256_checksum}\" \]\] && echo 'VERIFIED' || echo 'UNVERIFIED'"]
```

Comment:
What are you trying to achieve with this second checksum? Isn't the downloaded file already checksummed?

Reply:
I originally was verifying with gpg key, but the upstream distro used an inconsistent signature (!), so I’m using the distribution-provided checksum instead, which should be the same as the MacPorts-provided checksum, but does provide another independent check for this non-bootstrapped binary.

Reply:
That's my point – I don't think this actually gives us any additional assurance from a security PoV, and I don't really like using shell scripts inside Tcl, so I'd prefer if we just removed this check entirely.

Reply:
Okay, I'll delete it and add in gpg verification if stack fixes their signatures.